app/src/main/java/mozilla/lockbox/presenter/ItemListPresenter.kt

```diff
@@ -106,7 +108,25 @@ class ItemListPresenter(
             }
             .filterNotNull()
             .observeOn(AndroidSchedulers.mainThread())
-            .subscribe(view::setDisplayName)
+            .subscribe(view::setAccountName)
+            .addTo(compositeDisposable)
+
+        accountStore.profile
+            .map {
+                it.value?.email.asOptional()
+            }
+            .filterNotNull()
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe(view::setDisplayEmailName)
+            .addTo(compositeDisposable)
+
+        accountStore.profile
+            .map {
+                it.value?.avatar.asOptional()
+            }
+            .filterNotNull()
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe(view::setAvatarFromURL)
```

Comment:
This looks like three separate subscriptions to the same thing. Could you combine them?

e.g.

```kotlin
accountStore.profile
  .map { it.value?.asOptional() }
  .filterNotNull()
  .subscribe {
    view.avatarURL = it.avatar
    view.displayEmailName = it.email
    view.accountName = …
  }
```

Alternatively, you can use create a view model (not the Android architecture object) to pass all the objects up together.